commons/src/main/java/org/infinispan/commons/configuration/ClassWhiteList.java

```diff
@@ -45,23 +48,58 @@
 
    static {
       // Classes always allowed
+      // Primitives
+      SYS_ALLOWED_CLASSES.add(byte.class.getName());
+      SYS_ALLOWED_CLASSES.add(short.class.getName());
+      SYS_ALLOWED_CLASSES.add(int.class.getName());
+      SYS_ALLOWED_CLASSES.add(long.class.getName());
+      SYS_ALLOWED_CLASSES.add(float.class.getName());
+      SYS_ALLOWED_CLASSES.add(double.class.getName());
+      SYS_ALLOWED_CLASSES.add(char.class.getName());
+      SYS_ALLOWED_CLASSES.add(boolean.class.getName());
+
+      // Primitive Arrays
+      SYS_ALLOWED_CLASSES.add(byte[].class.getName());
+      SYS_ALLOWED_CLASSES.add(short[].class.getName());
+      SYS_ALLOWED_CLASSES.add(int[].class.getName());
+      SYS_ALLOWED_CLASSES.add(long[].class.getName());
+      SYS_ALLOWED_CLASSES.add(float[].class.getName());
+      SYS_ALLOWED_CLASSES.add(double[].class.getName());
+      SYS_ALLOWED_CLASSES.add(char[].class.getName());
+      SYS_ALLOWED_CLASSES.add(boolean[].class.getName());
+
+      // Boxed Primitives
       SYS_ALLOWED_CLASSES.add(Byte.class.getName());
       SYS_ALLOWED_CLASSES.add(Short.class.getName());
       SYS_ALLOWED_CLASSES.add(Integer.class.getName());
       SYS_ALLOWED_CLASSES.add(Long.class.getName());
       SYS_ALLOWED_CLASSES.add(Float.class.getName());
       SYS_ALLOWED_CLASSES.add(Double.class.getName());
-      SYS_ALLOWED_CLASSES.add(Number.class.getName());
       SYS_ALLOWED_CLASSES.add(Character.class.getName());
       SYS_ALLOWED_CLASSES.add(String.class.getName());
-      SYS_ALLOWED_CLASSES.add(HashSet.class.getName());
-      SYS_ALLOWED_CLASSES.add(HashMap.class.getName());
-      SYS_ALLOWED_CLASSES.add(Date.class.getName());
-      SYS_ALLOWED_CLASSES.add(Enum.class.getName());
       SYS_ALLOWED_CLASSES.add(Boolean.class.getName());
+
+      // Java.math
+      SYS_ALLOWED_CLASSES.add(BigInteger.class.getName());
+      SYS_ALLOWED_CLASSES.add(BigDecimal.class.getName());
+
+      // Java.time
+      SYS_ALLOWED_CLASSES.add(Instant.class.getName());
+      SYS_ALLOWED_CLASSES.add("java.time.Ser");
+
+
+      // Util
       SYS_ALLOWED_CLASSES.add(ArrayList.class.getName());
-      SYS_ALLOWED_CLASSES.add(Object[].class.getName());
-      SYS_ALLOWED_CLASSES.add("org.infinispan.jboss.marshalling.commons.JBossExternalizerAdapter");
+      SYS_ALLOWED_CLASSES.add(Date.class.getName());
+      SYS_ALLOWED_CLASSES.add(HashMap.class.getName());
+      SYS_ALLOWED_CLASSES.add(HashSet.class.getName());
```

Comment:
Apparently `HashSet` and `HashMap` are really bad:
https://gist.github.com/coekie/a27cc406fc9f3dc7a70d

Reply:
Ok, should we remove `ArrayList` as well? The rationale for adding that was convenience and the fact that those two were already allowed :smile:

Reply:
Yeah, I actually wonder about that regex as well, does that mean that any array of a user defined type can be used?

Reply:
The regex means that if a user white lists a class, say `Example`, they don't have to also add `Example[]` or `Example[][]` to the white list as well. However, if `Example` is not added to the list in the first place, it's not possible to marshall any of the variations.